app/models/customer.rb

```diff
@@ -3,4 +3,15 @@ class Customer < ApplicationRecord
 
   has_many :invoices
   has_many :transactions, through: :invoices
+  has_many :merchants, through: :invoices
+
+  def favorite_merchant
+    merchants
+      .joins(:invoices)
+      .joins(:transactions)
```

Comment:
This query looks great! If you want, you could combine lines 10 & 11 by doing this: `.joins(:invoices, :transactions)`